packages/passport-interface/src/EncryptedStorage.ts

```diff
@@ -48,3 +57,42 @@ export function isSyncedEncryptedStorageV3(
 ): storage is SyncedEncryptedStorageV3 {
   return storage._storage_version === "v3";
 }
+
+/**
+ * Deserialize a decrypted storage object and set up the PCDCollection and
+ * FeedSubscriptionManager to manage its data.  If the storage comes from
+ * an older format which doesn't include subscriptions, then the
+ * FeedSubscriptionManager will be empty.
+ */
+export async function deserializeStorage(
+  storage: SyncedEncryptedStorage,
+  pcdPackages: PCDPackage[]
+): Promise<{
+  pcds: PCDCollection;
+  subscriptions: FeedSubscriptionManager;
+}> {
+  let pcds: PCDCollection;
+  let subscriptions: FeedSubscriptionManager;
+
+  if (isSyncedEncryptedStorageV3(storage)) {
+    pcds = await PCDCollection.deserialize(pcdPackages, storage.pcds);
+    subscriptions = FeedSubscriptionManager.deserialize(
+      new NetworkFeedApi(),
+      storage.subscriptions
+    );
+  } else if (isSyncedEncryptedStorageV2(storage)) {
+    pcds = await PCDCollection.deserialize(pcdPackages, storage.pcds);
+    subscriptions = new FeedSubscriptionManager(new NetworkFeedApi());
```

Comment:
@robknight this is the lowest layer of my response to our discussion of syncing subscriptions.  Rather than returning null, I just create an empty manager so it's always available.  Let me know if you think this properly captures what we agreed.

Reply:
I agree in concept that we want default subscriptions never to be skipped.  I don't think they can go here directly due to dependency layering.  This is in passport-interface, while the definition of default subscriptions is in passport-client).  Does it belong there?  I notice that some of its data, like `zupassDefaultSubscriptions` comes from passport-interface, while other things like the server URL come from passport-client.
It looks like it's currently `addDefaultSubscriptions` is currently called everywhere subscriptions are downloaded or loaded, except in loadAfterLogin, which I guess waits for the first sync to do it.

I can see an option to pass in `addDefaultSubscriptions` as a function argument to here and other places to make sure callers don't miss it.  An alternative might be to have default subscriptions be added at the time that we prepare to poll subscriptions.  Assuming it's idempotent, that should be safe.  It could be a user-visible change if they view their subscriptions before the first attempt to fetch them, but that seems reasonable.

Reply:
I think I agree with you after having thought about it a bit more.